### PR TITLE
[sn-platform] Add necessary config to use function-mesh worker service

### LINKED ISSUE #575: [sn-platform] Use Function Mesh Worker Service
*Motivation*

Currently, the `sn-platform` uses the default Pulsar Function worker service with the k8s scheduler. However, there is a limitation on the existing Pulsar Function worker service. We should integrate the Function Mesh worker service as part of the `sn-platform`.

### PR DESCRIPTION
**Pull request by MarvinCai**
Fixes #575

### Motivation
Add necessary config to use function-mesh worker service

### Verifying this change

- [x] Make sure that the change passes the CI checks.
This change is a trivial rework / code cleanup without any test coverage.

### Documentation

Check the box below.

Need to update docs? 
- [x] `no-need-doc` 
  Function mesh worker service is transparent replacement of function service.

### CODE REVIEW

**Review by nlu90 — COMMENTED** [review]
What's this PR's relationship with my previous one: https://github.com/streamnative/charts/pull/636?

**Comment by MarvinCai** [comment]
`streamnative/sn-platform` is now purely operator based, only rely on Pulsar Operator CRDs

File: charts/sn-platform/templates/broker/broker-cluster.yaml [hunk 1]
```diff
@@ -281,6 +285,12 @@ spec:
 {{ toYaml . | indent 6 }}
 {{- end }}
       {{- if and (.Values.auth.vault.enabled) (.Values.auth.oauth.enabled) }}
+      {{- if and .Values.components.functions .Values.broker.functionmesh.enabled }}
+      # enable package service for function mesh
+      packagesManagementStorageProvider: "{{ .Values.broker.packagesManagement.storageProvider }}"
+      packagesReplicas: "{{ .Values.broker.configData.managedLedgerDefaultEnsembleSize }}"
```
Comment (fantapsody) [inline charts/sn-platform/templates/broker/broker-cluster.yaml:291]: Maybe we only need to configure this for now, and we can configure it in the operator.

**Comment by MarvinCai** [comment]
@fantapsody updated with latest operator version, PTAL

**Review by fantapsody — APPROVED** [review]

**Pull request merged**